### PR TITLE
echo support

### PR DESCRIPTION
**Pull request by aricart**
- Added `noEcho` option. When specified and true, messages published by the client won't be provided to the client's matching subscriptions. This feature requires server 1.2.0 or better. If the option is set on an older server the client emits an error and disconnects.

### CODE REVIEW

File: test/properties.ts [hunk 1]
```diff
@@ -119,4 +122,52 @@ test('configured options', async (t) => {
...
+        let sub2 = nc.subscribe(subj, ()=> {
+            c2++;
+        });
+        nc.publish(subj);
+        await nc.flush();
+        t.is(c2, 0);
+        nc.close();
+        lock.unlock();
+    }).catch((err) => {
+        if(err.code === ErrorCode.NO_ECHO_NOT_SUPPORTED) {
```
Comment (kozlovic) [inline test/properties.ts]: I assume this is in case the tests runs against an older server, right?

File: test/properties.ts [hunk 2]
```diff
@@ -119,4 +122,52 @@ test('configured options', async (t) => {
...
+    let server = new mockserver.ScriptedServer(0);
+    try {
+        await server.start();
+    } catch(ex) {
+        t.log(ex);
+    }
+    t.plan(1);
+    let nc = await connect({port: server.port, noEcho: true});
+    nc.on('error', (err) => {
+        t.is(err.code, ErrorCode.NO_ECHO_NOT_SUPPORTED);
```
Comment (kozlovic) [inline test/properties.ts]: This check that if there is an error, it is the expected one. But I don't see how this test ensure that the connect actually fails. In other words, there should be something that says on connect success, test fails.

File: test/properties.ts (hunk 1 above)
Comment (aricart) [reply test/properties.ts]: yeap

File: test/properties.ts (hunk 2 above)
Comment (kozlovic) [reply test/properties.ts]: We discussed and this is ok. In the implementation, the connect itself succeeds, but then the error "callback" is invoked. So this test makes sure that this is the case and so this is correct.

**Pull request merged**